libraries/AC_PID/AC_PID.cpp

```diff
@@ -120,6 +173,26 @@ void AC_PID::slew_limit(float smax)
     _slew_rate_max.set(fabsf(smax));
 }
 
+void AC_PID::set_notch_sample_rate(float sample_rate)
+{
+#if AC_PID_ADVANCED_ENABLED
+    if (!_adv_enable) {
+        return;
+    }
+
+    if (!is_zero(_notch_T_center_freq_hz.get()) &&
+        (!is_equal(sample_rate, _target_notch.sample_freq_hz())
+         || !is_equal(_notch_T_center_freq_hz.get(), _target_notch.center_freq_hz()))) {
+        _target_notch.init(sample_rate, _notch_T_center_freq_hz, _notch_bandwidth_hz, _notch_attenuation_dB);
```

Comment:
won't the init already not do anything if the center frequency is the same?

Reply:
No, it will get recalculated

Reply:
The notch filter does not actually record the sample rate, so it can't be done in the filter

Reply:
There is a variable for it...
https://github.com/ArduPilot/ardupilot/blob/daf427f6d612614f08a1ad770c42109f989e8dc5/libraries/Filter/NotchFilter.h#L45